File: app-ts/src/ZodObject/DataStore/AppSetting/AppSettingModel/CommentReciver/NiconicoLive/NiconicoLiveSettingModel.ts
import { z } from "zod";
export const NiconicoLiveSettingModel = z.object({
  配信URL: z
    .string()
    .describe("ニコ生の配信URLを入力してください")
    .default("test_niconico"),
  コメント受信: z
    .enum(["ON", "OFF"])
    .describe("コメント受信をONにするかOFFにするか選択してください")
    .default("OFF"),
  コメント禁止ワード: z
    .array(z.string())
    .describe("コメント禁止ワードを入力してください")
    .default(["4ね"]),
});
export type NiconicoLiveSettingModel = z.infer<typeof NiconicoLiveSettingModel>;

File: api/DataStore/AppSetting/AppSettingModel/CommentReciver/NiconicoLive/NiconicoLiveSettingModel.py


from enum import Enum
from pydantic import BaseModel, Field

class CommentOnOff(str,Enum):
    ON = "ON"
    OFF = "OFF"

class NiconicoLiveSettingModel(BaseModel):
    配信URL: str = Field(
        default="test_niconico", 
        title="ニコ生配信URL", 
        description="ニコ生の配信URLを入力してください"
        )
    コメント受信: CommentOnOff = Field(
        default=CommentOnOff.OFF,
        title="コメント受信", 
        description="コメント受信をONにするかOFFにするか選択してください"
    )
    コメント禁止ワード: list[str] = Field(
        default=["4ね"],
        title="コメント禁止ワード",
        description="コメント禁止ワードを入力してください"
    )